packages/client/tests/functional/referential-integrity/tests.ts

```diff
@@ -0,0 +1,677 @@
+          }),
+        ).resolves.toEqual(null)
+
+        expect(
+          prisma[profileModel].findUnique({
+            where: { userId: '2' },
+          }),
+        ).resolves.toEqual(null)
+
+        const profile2 = await prisma[profileModel].findUniqueOrThrow({
+          where: { userId: '3' },
+        })
+        expect(profile2).toEqual({
+          id: '2',
+          userId: '3',
+        })
+      })
+
+      test('[nested] update parent id [connect] child should succeed if the relationship already existed', async () => {
+        await prisma[userModel].create({
+          data: {
+            id: '1',
+            profile: {
+              create: { id: '1' },
+            },
+          },
+        })
+        await prisma[userModel].create({
+          data: {
+            id: '2',
+            profile: {
+              create: { id: '2' },
+            },
+          },
+        })
+
+        // TODO: is this a bug? Probably it's trying to disconnect under the hood?
+        try {
+          await prisma[userModel].update({
+            where: { id: '1' },
+            data: {
+              id: '3',
+              profile: {
+                connect: { id: '1' },
+              },
+            },
+          })
+          assertUnreachable()
+        } catch (error) {
+          expect(error.message).toContain('The change you are trying to make would violate the required relation \'ProfileOneToOneToUserOneToOne\' between the `ProfileOneToOne` and `UserOneToOne` models.')
+        }
+      })
+
+      test('[nested] update parent id [disconnect] child should succeed', async () => {
+        await prisma[userModel].create({
+          data: {
+            id: '1',
+            profile: {
+              create: { id: '1' },
+            },
+          },
+        })
+        await prisma[userModel].create({
+          data: {
+            id: '2',
+            profile: {
+              create: { id: '2' },
+            },
+          },
+        })
+
+        try {
+          await prisma[userModel].update({
+            where: { id: '1' },
+            data: {
+              id: '3',
+              profile: {
+                disconnect: true,
+              },
+            },
+          })
+          assertUnreachable()
+        } catch (error) {
+          expect(error.message).toContain('The change you are trying to make would violate the required relation \'ProfileOneToOneToUserOneToOne\' between the `ProfileOneToOne` and `UserOneToOne` models.')
+        }
+      })
+
+      test('[nested] update parent id [update] child should succeed', async () => {
+        await prisma[userModel].create({
+          data: {
+            id: '1',
+            profile: {
+              create: { id: '1' },
+            },
+          },
+        })
+        await prisma[userModel].create({
+          data: {
+            id: '2',
+            profile: {
+              create: { id: '2' },
+            },
+          },
+        })
```

Comment:
We should remove these copied and pasted lines everywhere and use a small function that does the same.
It would be easier to see what we actually want to test in these tests, and remove potential copied and pasted typos.